c7n/resources/vpc.py

```diff
@@ -1469,6 +1471,83 @@ class resource_type(object):
         taggable = False
 
 
+@NetworkAddress.action_registry.register('release')
+class AddressRelease(BaseAction):
+    """Action to release elastic IP address(es)
+
+    Use the force option to cause any attached elastic IPs to
+    also be released.  Otherwise, only unattached elastic IPs
+    will be released.
+
+    You can filter by AllocationId, if needed
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: release-network-addr
+                resource: network-addr
+                filters:
+                  - AllocationId: ...
+                actions:
+                  - type: release
+                    force: true|false
+    """
+
+    # Schema, permissions, method process
+    schema = type_schema('release', force={'type': 'boolean'})
+    permissions = ('ec2:ReleaseAddress','ec2:DisassociateAddress',)
+
+    def process_attached(self, client, associated_addrs, force=True):
+        def msg_assoc_extractor(x):
+            return ('instance', x['InstanceId']) if 'InstanceId' in x else \
+                ('network interface', x.get('NetworkInterfaceId'))
+
+        log_template = 'Elastic IP {0} is attached to {1} {2} and will not be released.  Specify \'force: "true"\' to release'  # noqa: E501
+
+        detached_cnt = 0
+        deassoc_addrs = []
+        for aa in associated_addrs:
+            if force:
```

Comment:
minor, but stylistically its more readable to dedent code where possible, ie in this case you have a big conditional block around force condition, handling the if not force case in a conditional would allow dedenting this block.